mne/decoding/time_gen.py

```diff
@@ -641,9 +687,19 @@ def _predict(X, estimators):
     from scipy import stats
     from sklearn.base import is_classifier
     # Initialize results:
-    n_epochs = X.shape[0]
+
+    orig_shape = X.shape
+    n_epochs = orig_shape[0]
+    n_times = orig_shape[-1]
+
     n_clf = len(estimators)
 
+    # in simple case, we are predicting each time sample as if it
+    # was a different epoch
+    if is_single_time_sample:  # treat times as trials for optimization
+        X = np.hstack(X).T
+    n_epochs_tmp = len(X)
+
```

Comment:
treat times as trials within CV or for all data, depending on call -- reshape data for that.